File: test_assign.js
const fetch = require('node-fetch');

async function test() {
    const loginRes = await fetch('http://localhost:8000/api/v1/auth/login', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ email: 'admin@campus.edu', password: 'admin' })
    });
    const { access_token } = await loginRes.json();

    const res = await fetch('http://localhost:8000/api/v1/tasks/assign', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${access_token}`
        },
        body: JSON.stringify({
            issue_id: 'REQ-1002',
            asset_id: 'asset-4',
            assigned_to: 'u-service-1',
            assigned_by: 'u-admin-1',
            priority: 'medium'
        })
    });

    const text = await res.text();
    console.log('Status:', res.status);
    console.log('Response:', text);
}

test();

File: test_assign.py
import urllib.request as r
import json

base = "http://localhost:8000/api/v1"

try:
    req = r.Request(f"{base}/auth/login", data=json.dumps({"email": "admin@campus.edu", "password": "password"}).encode('utf-8'), headers={"Content-Type": "application/json"})
    toks = json.loads(r.urlopen(req).read())
    token = toks["access_token"]
    user_id = toks["user"]["id"]
except Exception as e:
    print("Login Failed:", e)
    if hasattr(e, 'read'):
        print(e.read().decode())
    exit(1)

try:
  req2 = r.Request(f"{base}/tasks/assign", data=json.dumps({
      "issue_id": "REQ-1002",
      "asset_id": "asset-4",
      "assigned_to": "u-service-1", 
      "assigned_by": user_id,
      "priority": "medium"
  }).encode('utf-8'), headers={"Content-Type": "application/json", "Authorization": f"Bearer {token}"})
  print("Success:", r.urlopen(req2).read().decode())
except Exception as e:
  print("Assign Failure:", e)
  if hasattr(e, 'read'):
      print(e.read().decode())
